Adds new personal goal for scientists

CSCMe opened the pull request:
Adds new "Don't let a slime attack you" personal goal for scientists
Also adds a framework for failable achievement-type personal goals
Fixes job-specific personal goals not generating after spawning (wasn't noticed because there were no job-specific goals before)

🆑 CSCMe
rscadd: Adds "Don't let a slime snack on you" personal goal for scientists.
/🆑

quardbreak commented on code/modules/goals/_goal.dm:
```diff
@@ -54,5 +55,11 @@
+	if(failure_message && !check_success())
+		if(istype(owner, /datum/mind))
+			var/datum/mind/mind = owner
+			to_chat(mind.current, "<font color='red'><b>[failure_message]</b></font>")
```
should be SPAN_DANGER/WARNING macro, so it will respect user's chat darkmode

SierraKomodo replied:
`SPAN_DANGER()` would match the bold red in the current implementation.

SierraKomodo commented on code/modules/goals/_goal.dm:
```diff
@@ -54,5 +55,11 @@
 			to_chat(mind.current, "<font color='green'><b>[completion_message]</b></font>")
 
+/datum/goal/proc/on_failure()
+	if(failure_message && !check_success())
+		if(istype(owner, /datum/mind))
+			var/datum/mind/mind = owner
+			to_chat(mind.current, "<font color='red'><b>[failure_message]</b></font>")
```
The if statements can be merged together.

```suggestion
	if(failure_message && !check_success() && istype(owner, /datum/mind))
		var/datum/mind/mind = owner
		to_chat(mind.current, "<font color='red'><b>[failure_message]</b></font>")
```

CSCMe replied:
done 👍

SierraKomodo commented on code/modules/goals/goal_mind.dm:
```diff
@@ -28,6 +28,8 @@
 		return FALSE
 
 	var/list/available_goals = SSgoals.global_personal_goals ? SSgoals.global_personal_goals.Copy() : list()
+	if(job && LAZYLEN(job.possible_goals))
+		available_goals |= job.possible_goals
```
Why is this being moved?

CSCMe replied:
Because before it was only called if the add_amount was null.
This was only the case when the goals are generated at spawn, meaning that job-goals would only generate at spawn.
The move makes it so job-goals generate when clicking "reroll goal" or "generate new goal" as well

It didn't seem intentional and there weren't any job-goals before, so I changed it

SierraKomodo reviewed (changes requested):
This doesn't appear to actually be assigning personal goals to scientists when joining the round.

CSCMe commented:
What do you mean exactly? 
Do you mean it doesn't assign goals to scientists in general or just the new personal goal?

SierraKomodo commented:
> What do you mean exactly?
> Do you mean it doesn't assign goals to scientists in general or just the new personal goal?

It's not assigning any personal goals at all, only the departmental goals.

CSCMe commented:
Oh alright.
I was checking it on my PC just now and it seems to work.
Did you set the "Give Personal Goals" preference away from "Never"? It's the default value and it doesn't tell you in the goal interface

CSCMe committed:
Adds new personal goal for scientists

Adds failable achievement-type personal goals

Fixes job-specific personal goals not generating after initial spawn

Fixes job-specific min and max personal goal add amounts requiring job-specific personal goals to be used

Adds a scientist-specific "don't get snacked by slime" personal goal

Domic2 commented:
Can confirm, goal spawn is a-ok!

CSCMe commented:
Hey sorry if this is annoying and you're already aware but this is ready for review ^^
